src/Examples/UnitModels/Operations/skeleton_unit_testing.ipynb

```diff
@@ -0,0 +1,420 @@
+    "from idaes.core import FlowsheetBlock\n",
+    "from idaes.models.unit_models import SkeletonUnitModel\n",
+    "from idaes.core.util.model_statistics import degrees_of_freedom\n",
+    "from pyomo.util.check_units import assert_units_consistent"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "For demonstrative purposes, we will build a simple model manually defining state variables relations for two generic components. In this model, the variables specify an FcTP system where molar flow of each component, temperature and pressure are selected as state variables. Users can use this structure to write custom relations between inlet and outlet streams; for example, defining outlet pressure using an explicit correlation or outlet flow of a specific component from a known accumulation or loss in the system."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "m = ConcreteModel()\n",
+    "m.fs = FlowsheetBlock(default={\"dynamic\": False})\n",
+    "\n",
+    "m.fs.skeleton = SkeletonUnitModel(default={\"dynamic\": False})\n",
+    "m.fs.skeleton.comp_list = Set(initialize=[\"c1\", \"c2\"])  # two generic components\n",
+    "\n",
+    "# input vars for skeleton\n",
+    "# m.fs.time is a pre-initialized Set belonging to the FlowsheetBlock; for dynamic=False, time=[0]\n",
+    "m.fs.skeleton.flow_comp_in = Var(m.fs.time, m.fs.skeleton.comp_list, initialize=1.0, units=pyunits.mol/pyunits.s)\n",
+    "m.fs.skeleton.temperature_in = Var(m.fs.time, initialize=298.15, units=pyunits.K)\n",
+    "m.fs.skeleton.pressure_in = Var(m.fs.time, initialize=101, units=pyunits.kPa)\n",
+    "\n",
+    "# output vars for skeleton\n",
+    "m.fs.skeleton.flow_comp_out = Var(m.fs.time, m.fs.skeleton.comp_list, initialize=1.0, units=pyunits.mol/pyunits.s)\n",
+    "m.fs.skeleton.temperature_out = Var(m.fs.time, initialize=298.15, units=pyunits.K)\n",
+    "m.fs.skeleton.pressure_out = Var(m.fs.time, initialize=101, units=pyunits.kPa)\n",
+    "\n",
+    "# Surrogate model equations\n",
+    "\n",
+    "# flow equation\n",
+    "def rule_flow(m, t, i):\n",
+    "    return m.flow_comp_out[t, i] == m.flow_comp_in[t, i]\n",
+    "m.fs.skeleton.eq_flow = Constraint(m.fs.time, m.fs.skeleton.comp_list, rule=rule_flow)\n",
+    "\n",
+    "# note that all terms need to have consistent units - the \"10\" terms could be defined as separate variables with units\n",
+    "m.fs.skeleton.eq_temperature = Constraint(expr=m.fs.skeleton.temperature_out[0]\n",
+    "                                          == m.fs.skeleton.temperature_in[0] + 10*pyunits.K)\n",
+    "m.fs.skeleton.eq_pressure = Constraint(expr=m.fs.skeleton.pressure_out[0]\n",
+    "                                       == m.fs.skeleton.pressure_in[0] - 10*pyunits.kPa)\n",
+    "\n",
+    "# dictionaries relating state properties to custom variables\n",
```

Comment:
This part especially needs more description - what are these `dicts`, how they will be used and what they must contain.